libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -37,27 +37,16 @@
 #include <set>
 #include <string>
 #include <memory>
-#include <vector>
+#include <queue>
 
 namespace solidity::frontend
 {
 
 class YulUtilFunctions;
 class ABIFunctions;
 
-struct AscendingFunctionIDCompare
-{
-	bool operator()(FunctionDefinition const* _f1, FunctionDefinition const* _f2) const
-	{
-		// NULLs always first.
-		if (_f1 != nullptr && _f2 != nullptr)
-			return _f1->id() < _f2->id();
-		else
-			return _f1 == nullptr;
-	}
-};
-
-using DispatchSet = std::set<FunctionDefinition const*, AscendingFunctionIDCompare>;
+using DispatchQueue = std::queue<FunctionDefinition const*>;
```

Comment:
I used a `queue` here so we can have constant time processing the functions for codegen in FIFO order, and I kept the `set` for logarithmic lookup in the `m_internalDispatchMap`, but maybe this could also be replace by a queue or a vector, but then we would have linear lookup.